File: frontend/src/types/document.ts
export type DocumentStatus =
  | 'pending'
  | 'processing'
  | 'completed'
  | 'failed'

export interface Document {
  id: number
  knowledge_base_id: number
  name: string
  file_type: string
  file_path: string
  file_size: number
  status: DocumentStatus
  error_message: string | null
  created_at: string
  updated_at: string
}

File: backend/app/constants/document.py
from enum import Enum

class DocumentStatus(str, Enum):
    """
    文档处理状态。

    📌 必懂：状态机的核心定义。
    """

    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"